### DISCUSSION TITLE
Custom Themes Overhaul

### DISCUSSION THREAD

**Discussion by kgar** [kgar/foundry-vtt-tidy-5e-sheets#110] [Ideas]
I'm adding this discussion item now and will put more details in later.
inspired by some discussion in: https://github.com/kgar/foundry-vtt-tidy-5e-sheets/issues/108

After Tidy 5e Sheets rewrite goes v1 and becomes the official Foundry package, I intend, at some point, to overhaul the theme settings feature.

Some things I've tossed around so far:
- GM (world scoped) / User (client scoped) can add whole themes to the theme dropdown
- Individual sheets can choose a specific theme (my main PC wizard has a cool arcane theme, my PC artificer has icier/futuristic colors, etc.)
- Theme Settings UI allows the GM to actually change the default dark and default light themes for everyone
- A fuller, more comprehensive theme roller
- A github repository of user-submitted themes which could be either imported into a world via JSON file, one-by-one, or maybe even referenced by a search dialog
- Expose the most common color pickers for the average user; provide advanced mode for all variables are displayed for edit.

## Risks

There will likely need to be a data migration from current to new. Doing these responsibly requires care. Alternatively, it could be a one-time macro, but that's more of an Alpha testing approach.

Providing so much flexibility could lead to questions of how to apply these styles to integrating modules, which could be tricky and/or finicky.

**Comment by Dyrnwyrm** [comment]
Thanks for answering and yeah i was thinking on a way to let the player pick their sheet colors, all of my players love the dark sheet since the O.G Tidy release, but a lot of them want custom hp bars, custom accents and stuff like that, maybe a nice way to do this would be let the player save their edits and the GM uploads the Json file to the server, also the whole github repository is a really cool idea, you can have a lot of user created themes as you say players love to have custom sheets since a lot of the time they are looking at the scene with the sheet open so having that extra style helps them feel immersed in the game.

**Reply by kgar** [reply]
I meant to ask: are they able to access the current Theme Settings?

As any user, you should be able to change theme settings for your user account:

![image](https://github.com/kgar/foundry-vtt-tidy-5e-sheets/assets/4632129/38bb0cc7-a4e0-489f-8f76-e76e7d6e4345)

Here, I am logged in as a player and changed my HP bar to orange:

![image](https://github.com/kgar/foundry-vtt-tidy-5e-sheets/assets/4632129/580602e4-7df7-4771-b328-415ee7ce7515)

I wasn't sure based on our conversation so far. I know we're also talking about additional stuff on top of that, but I wanted to make sure the players knew about this feature in the rewritten sheets.
